src/stores/workspace/WorkspaceStore.ts

```diff
@@ -157,23 +157,63 @@ export default class WorkspaceStore {
 
 	@action
 	public onSavedItemSelect = (savedItem: EventTreeNode | EventAction | EventMessage) => {
-		if (this.workspacesStore.searchWorkspace === this) {
-			const newWorkspace = this.workspacesStore.createWorkspace({
+		if (this !== this.workspacesStore.searchWorkspace) {
+			if (isEventMessage(savedItem)) {
+				this.viewStore.activePanel = this.messagesStore;
+				this.messagesStore.onMessageSelect(savedItem);
+			} else {
+				this.viewStore.activePanel = this.eventsStore;
+				this.eventsStore.onEventSelect(savedItem);
+			}
+			return;
+		}
+
+		let newWorkspace: WorkspaceStore;
+		new Promise<void>(res => {
+			newWorkspace = this.workspacesStore.createWorkspace({
 				entity: savedItem,
 				layout: isEventMessage(savedItem) ? [0, 100] : [100, 0],
 			});
-
 			this.workspacesStore.addWorkspace(newWorkspace);
-			return;
-		}
+			res();
+		}).then(() => {
+			if (isEventMessage(savedItem)) {
+				const filterStore = newWorkspace.messagesStore.filterStore;
+				filterStore.setMessagesFilter(
+					filterStore.messagesFilter,
+					this.searchStore.savedSseMessagesFilters || {
```

Comment:
So you are saving last applied filters in search panel and initializing Messages or Events panel with them. Hovewer `onSavedItemSelect` is also invoked on BookmarkItem click or on Item that was found in GraphSearch. And there is no guarantee that those filters and item have connections. We might want to have a separate handler for SearchPanel items. And those should be initialized with currentSearch.request.filters since every request has its own saved filter